### ISSUE TITLE
Open-ended ranges and rangeUntil operator

### ISSUE THREAD

**Issue by ilya-g** [Kotlin/KEEP#314]
This issue is for discussion of [the proposal](https://github.com/Kotlin/KEEP/blob/master/proposals/open-ended-ranges.md) to introduce `rangeUntil` (`..<`) operator in Kotlin and the corresponding types for open-ended ranges.

Proposal PR #315

**Comment by Whathecode** [comment]
Thank you for this excellent overview of planned additions and tradeoffs considered!

I want to comment on the following:

> **Common supertype of ClosedRange and OpenEndRange**
> ...
> Introducing a more dedicated common supertype of the ClosedRange and OpenEndRange could help in this situation, however, it's unclear what useful operations such supertype would provide.

I'm the library author of `kotlinx.interval`, which approaches ranges using a common generic base type (including a potentially separate type for range 'sizes'). To help assess whether or not such a base type may be useful, I will soon continue porting some of the existing C# implementation I did to Kotlin, and add example use cases.

In short, I remember using them in a generic binary sort algorithm, and operations on range _collections_. I.e., given a collection of time intervals, does a specified timestamp fall within any of the ranges? But, likely those could have been simple collections of closed ranges only. I don't recall fully whether mixing half-open and closed ranges was ever a true requirement for the latter use case.

**Comment by zarechenskiy** [comment]
Closing the issue as the feature was graduated to Stable in Kotlin 1.9.0
